server/lobby.js

```diff
@@ -392,6 +392,29 @@ class Lobby {
             return;
         }
 
+        if(gameDetails.quickJoin) {
+            let sortedGames = _.sortBy(this.games, 'createdAt');
+            let gameToJoin = sortedGames.find(game => !game.started && game.gameType === gameDetails.gameType && _.size(game.players) < 2 && !game.password);
+
+            if(gameToJoin) {
+                gameToJoin.join(socket.id, socket.user.getDetails(), undefined, (err, message) => {
+                    if(err) {
+                        socket.send('passworderror', message);
```

Comment:
This doesn't seem like the proper message to send back since passworded games were explicitly excluded in the `gameToJoin` array.

Reply:
There is no generic error mechanism, this check is actually redundant as the only exceptions thrown are related to password checking.  However, I'll leave this in for now as we need a generic error mechanism to prevent things like a bug Zack just reported with joining blocked games